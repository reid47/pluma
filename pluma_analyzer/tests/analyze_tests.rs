#[macro_use]
mod macros;

test_analyze! {
  undefined_identifier_simple (false): r#"
    |wat
  "#,

  defined_identifier_simple (true): r#"
    |let fine = 47
    |
    |fine
  "#,

  intrinsic_def_binary_op_valid (true): r#"
    |intrinsic_type Int
    |intrinsic_def Int + Int -> Int
    |
    |let result = 3 + 4
  "#,

  intrinsic_def_binary_op_invalid (false): r#"
    |intrinsic_type Int
    |intrinsic_def Int + Int -> Int
    |
    |let result = 3 + "yikes"
  "#,
}
