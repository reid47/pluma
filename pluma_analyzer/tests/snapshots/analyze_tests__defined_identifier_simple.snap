---
source: pluma_analyzer/tests/analyze_tests.rs
expression: "let fine = 47\n\nfine"
---

=== Source ===
let fine = 47

fine

=== Top-level scope ===
Scope {
    levels: [
        ScopeLevel {
            bindings: {
                "fine": Binding {
                    typ: Int,
                    ref_count: 1,
                    pos: (
                        4,
                        8,
                    ),
                    kind: Let,
                },
            },
        },
    ],
    type_bindings: {},
}

